src/libpmem2/badblocks_ndctl.c

```diff
@@ -549,9 +549,10 @@ pmem2_badblock_next(struct pmem2_badblock_context *bbctx,
 	unsigned e;
 	int ret;
 
+	*bb = BB_ZERO;
```

Comment:
what is the bug? This variable should not be read if return status is `!= 0`. I'm guessing it's a static analysis problem, and I would suggest just adding an ASSERT somewhere to prevent it.